plugin/evm/vm.go

```diff
@@ -201,17 +199,10 @@ func init() {
 	// Preserving the log level allows us to update the root handler while writing to the original
 	// [os.Stderr] that is being piped through to the logger via the rpcchainvm.
 	originalStderr = os.Stderr
-
-	// Register the codec for the atomic block sync summary
-	var err error
-	networkCodec, err = message.NewCodec(atomicsync.AtomicSyncSummary{})
-	if err != nil {
-		panic(fmt.Errorf("failed to create codec manager: %w", err))
-	}
 }
 
-// VM implements the snowman.ChainVM interface
-type VM struct {
+// sharedEvm implements the snowman.ChainVM interface
+type sharedEvm struct {
```

Comment:
Could we put this back? "VM" seems sufficient and as the wrapped VM is in another package it seems that we can leave this.

Reply:
I dont want to export this as without default extensions it will break. we can check if required extensions are nil in initialize to make sure it won't break if we want to. but I still slightly think we should not directly expose vm.

Reply:
I don't think `sharedEvm` is a good name here (what is it shared between? runtime? compile time?) Perhaps we can use a more descriptive name or `vm`?

 I don't disagree that unexported is better, but renaming this seems like it should be its own PR as it is not necessary for this PR and makes this PR difficult to follow.